src/ln/channelmanager.rs

```diff
@@ -677,6 +677,15 @@ impl ChannelManager {
 	/// Call this upon creation of a funding transaction for the given channel.
 	/// Panics if a funding transaction has already been provided for this channel.
 	pub fn funding_transaction_generated(&self, temporary_channel_id: &Uint256, funding_txo: OutPoint) {
+
+		macro_rules! add_pending_event {
```

Comment:
nit: maybe move down closer to first use?

Reply:
Might as well fix this while you're fixing other stuff, so the duplicate-lock issue doesn't get re-introduced.